lib/castle/client.rb

```diff
@@ -40,6 +50,7 @@ def identify(options = {})
       options = Castle::Utils.deep_symbolize_keys(options || {})
 
       return unless tracked?
+      options[:timestamp] = @timestamp if @timestamp
```

Comment:
I'd change it for all methods to `options[:timestamp] ||= ` that way you can also pass in a timestamp to identify/track